### ISSUE TITLE
The quad example panics with the OpenGL backend

### ISSUE THREAD

**Issue by vainiovano** [gfx-rs/gfx#2754]
Short info header:
- GFX version: git
- OS: Debian GNU/Linux sid
- GPU: Mesa DRI Intel(R) HD Graphics 520 (Skylake GT2)

The type of the backbuffer is FrameBuffer so `frame_images` is set to an empty Vec.

https://github.com/gfx-rs/gfx/blob/42bb2065306a6a438cb7baf1d16af9fb0949f083/examples/quad/main.rs#L422
https://github.com/gfx-rs/gfx/blob/42bb2065306a6a438cb7baf1d16af9fb0949f083/examples/quad/main.rs#L449
https://github.com/gfx-rs/gfx/blob/42bb2065306a6a438cb7baf1d16af9fb0949f083/examples/quad/main.rs#L450

Then `image_acquire_semaphores` is not filled with any data.

https://github.com/gfx-rs/gfx/blob/42bb2065306a6a438cb7baf1d16af9fb0949f083/examples/quad/main.rs#L456-L462
https://github.com/gfx-rs/gfx/blob/42bb2065306a6a438cb7baf1d16af9fb0949f083/examples/quad/main.rs#L490-L496

And finally the program panics on trying to index the empty `image_acquire_semaphores`

https://github.com/gfx-rs/gfx/blob/42bb2065306a6a438cb7baf1d16af9fb0949f083/examples/quad/main.rs#L743-L747

**Comment by msiglreith** [comment]
Thanks for tracking it down!
It would be resolved once https://github.com/gfx-rs/gfx/pull/2663 is merged, as it would remove the Backbuffer::Framebuffer variant which GL uses currently.

**Comment by icefoxen** [comment]
Entertainingly, I was just about to submit the exact same thing.  :grin: I tested that PR and the quad example works with the OpenGL backend, using Linux/X11 with a GeForce GTX 1060 and official NVidia drivers.

**Comment by AregevDev** [comment]
[Windows] panics when resizing the `quad` example to 0 height with GL backend

**Comment by kvark** [comment]
Fixed by #2663
@AregevDev please file a separate issue for this
